File: src/main/java/com/handwoong/everyonewaiter/service/MemberService.java
package com.handwoong.everyonewaiter.service;

import com.handwoong.everyonewaiter.dto.MemberDto;
import com.handwoong.everyonewaiter.dto.OnlyMsgResDto;
import java.util.List;

public interface MemberService {

    Long register(MemberDto.RequestDto memberDto);

    MemberDto.ResponseDto findMember(Long memberId);

    MemberDto.ResponseDto findMemberByUsername(String username);

    List<MemberDto.ResponseDto> findMemberList();

    OnlyMsgResDto changePassword(String username, MemberDto.PwdRequestDto passwordDto);

    OnlyMsgResDto deleteMember(String username, MemberDto.PwdRequestDto passwordDto);
}

File: src/main/kotlin/com/handwoong/everyonewaiter/service/member/MemberService.kt
package com.handwoong.everyonewaiter.service.member

import com.handwoong.everyonewaiter.dto.member.*

interface MemberService {

    fun login(memberDto: MemberLoginRequest): TokenResponse

    fun register(memberDto: MemberCreateRequest)

    fun findMemberById(memberId: Long): MemberResponse

    fun findMemberByUsername(username: String): MemberResponse

    fun findAllMemberList(): List<MemberResponse>

    fun changePassword(username: String, passwordDto: PasswordRequest)

    fun deleteMember(username: String, passwordDto: PasswordRequest)

}
